### PR TITLE
Synchronize shutdown hooks

### LINKED ISSUE #73: ConcurrentModificationException thrown
Just noticed a concurrent modification issue when attempting to shutdown LL:

```
Slf4jLogger started
Unix ready on /var/run/landlord/landlord.sock
An internal error has occurred within landlord given process 4. Stacktrace follows.
java.util.ConcurrentModificationException
	at java.util.IdentityHashMap$IdentityHashMapIterator.nextIndex(IdentityHashMap.java:735)
	at java.util.IdentityHashMap$KeyIterator.next(IdentityHashMap.java:826)
	at scala.collection.convert.Wrappers$JIteratorWrapper.next(Wrappers.scala:40)
	at scala.collection.Iterator.foreach(Iterator.scala:944)
	at scala.collection.Iterator.foreach$(Iterator.scala:944)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1432)
	at scala.collection.IterableLike.foreach(IterableLike.scala:71)
	at scala.collection.IterableLike.foreach$(IterableLike.scala:70)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
	at scala.collection.TraversableLike.filterImpl(TraversableLike.scala:247)
	at scala.collection.TraversableLike.filterImpl$(TraversableLike.scala:245)
	at scala.collection.AbstractTraversable.filterImpl(Traversable.scala:104)
	at scala.collection.TraversableLike.filter(TraversableLike.scala:259)
	at scala.collection.TraversableLike.filter$(TraversableLike.scala:259)
	at scala.collection.AbstractTraversable.filter(Traversable.scala:104)
	at com.github.huntc.landlord.JvmExecutor$.removeShutdownHooks(JvmExecutor.scala:152)
	at com.github.huntc.landlord.JvmExecutor$$anonfun$starting$1$$anon$1.$anonfun$uncaughtException$1(JvmExecutor.scala:391)
	at java.lang.Thread.run(Thread.java:748)
RESOURCE RETENTION - Process 4 has 12 thread(s) still running - check that your shutdown hooks and/or `trap` handler is shutting down all that it needs to.
Remaining threads: mqtt-gateway-akka.actor.default-dispatcher-2, metrics-reporting, mqtt-gateway-scheduler-1, mqtt-gateway-akka.actor.default-dispatcher-3, chronicle-weak-reference-cleaner, mqtt-gateway-akka.actor.default-dispatcher-18, mqtt-gateway-akka.io.pinned-dispatcher-6, mqtt-gateway-akka.actor.default-dispatcher-4, mqtt-gateway-akka.actor.default-dispatcher-10, mqtt-gateway-akka.actor.default-dispatcher-19, mqtt-gateway-akka.actor.default-dispatcher-5, MQTT Con: mqtt-gateway-2063380839804838
Coordinated shutdown phase [before-actor-system-terminate] timed out after 5000 milliseconds
```

### PR DESCRIPTION
**Pull request by huntc**
This PR avoids concurrent execution errors when gathering the list of shutdown hooks.

Fixes #73

### CODE REVIEW

File: landlordd/daemon/src/main/scala/com/github/huntc/landlord/JvmExecutor.scala [hunk 1]
```diff
@@ -133,39 +133,41 @@ object JvmExecutor {
    * `IllegalStateException` is thrown.
    */
   private[landlord] def removeShutdownHooks(group: ThreadGroup): Set[Thread] = {
-    var hooks: Set[Thread] = null
+    val applicationShutdownHooksClass = Class.forName("java.lang.ApplicationShutdownHooks")
```
Comment (dwijnand) [inline landlordd/daemon/src/main/scala/com/github/huntc/landlord/JvmExecutor.scala]: Note that this runtime lookup can fail. Though looks like any failure that might've occurred wasn't being handled before either, when it was in the try block. (Is that a mistake?)

I wonder: any reason this isn't just a class literal `classOf[java.lang.ApplicationShutdownHooks]`?

Comment (huntc) [reply landlordd/daemon/src/main/scala/com/github/huntc/landlord/JvmExecutor.scala]: Yeah, I agree. I'll tighten it up.

Comment (huntc) [reply landlordd/daemon/src/main/scala/com/github/huntc/landlord/JvmExecutor.scala]: Thanks @dwijnand - I've now tightened up the handling.

File: landlordd/daemon/src/main/scala/com/github/huntc/landlord/JvmExecutor.scala [hunk 2]
```diff
@@ -133,41 +133,44 @@ object JvmExecutor {
...
-        Runtime.getRuntime.removeShutdownHook(hook)
       }
+    } catch {
+      case _: ClassNotFoundException | _: ExceptionInInitializerError | _: LinkageError =>
+        throw new IllegalStateException("Could not access java.lang.ApplicationShutdownHooks")
+    }
 
-      hooks
+    try {
+      hooks.foreach(Runtime.getRuntime.removeShutdownHook)
```
Comment (huntc) [inline landlordd/daemon/src/main/scala/com/github/huntc/landlord/JvmExecutor.scala]: Important to do this outside of the synchronized block given the potential for deadlocks.

**Pull request merged**